plugins/modules/luks_device.py

```diff
@@ -523,6 +575,29 @@ def is_luks(self, device):
         result = self._run_command([self._cryptsetup_bin, 'isLuks', device])
         return result[RETURN_CODE] == 0
 
+    def get_luks_type(self, device):
+        ''' get the luks type of a device
+        '''
+        if self.is_luks(device):
+            with open(device, 'rb') as f:
+                f.seek(LUKS2_HEADER_OFFSETS[0])
+                data = f.read(LUKS_HEADER_L)
+                if data == LUKS2_HEADER2:
+                    return 'luks2'
+                else:
+                    return 'luks1'
+        return None
+
+    def is_luks_slot_set(self, device, keyslot):
+        ''' check if a keyslot is set
+        '''
+        result = self._run_command([self._cryptsetup_bin, 'luksDump', device])
```

Comment:
I really wish `--dump-json-metadata` would also be available for LUKS1. That way extracting this information would be a lot easier...